build_runner/lib/src/server/server.dart

```diff
@@ -39,8 +44,24 @@ class ServeHandler implements BuildState {
 
   FutureOr<Response> _handle(Request request, String rootDir) async {
     await currentBuild;
+    if (_lastBuildResult.status == BuildStatus.failure) {
+      return new Response(200,
```

Comment:
should this be 200? We don't want to indicate a server error?

Reply:
Ya I went back and forth on this a bit but it looks like chrome/firefox at least will render the page even if the response status is 500.

Updated to `HttpStatus.INTERNAL_SERVER_ERROR`.